### PR TITLE
Fix warning icon

### LINKED ISSUE #444: Warning icons don't show up on Windows
This may be behind some of the mysterious "camera sources not working" problems on Windows.  The exception button uses fontawesome, which doesn't work on windows.

icon

This should be fixed to use a .png image ASAP so people can actually see errors.

### PR DESCRIPTION
**Pull request by ThomasJClark**
It should actually render on Windows now, and it looks better

Closes #444

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
Tests fail.

**Comment by JLLeitschuh** [comment]
Tests still fail.

**Comment by JLLeitschuh** [comment]
### Before
Before Image

### After
After Image

1. I liked how the button looked before.
2. Why is there a vertical bar there, even when the alert isn't. It isn't dividing anything when the alert isn't there.
3. The tests still fail.

**Comment by ThomasJClark** [comment]
> I liked how the button looked before.

I agree that the previous version of the PR in your screenshot doesn't look that great, but the old button was terrible.  The icon was way too small to tell what it is, and the button isn't even the same height as the buttons around it.

I updated the PR based on this feedback.  The tests all pass on my machine.

**Comment by ThomasJClark** [comment]
There's one test that keeps throwing an exception on Travis but not my machine.  I'm gonna see what happens when if I just remove that test, since it seems pretty unnecessary.

File: ui/src/test/java/edu/wpi/grip/ui/components/ExceptionWitnessResponderButtonTest.java [hunk 1]
```diff
@@ -28,7 +27,7 @@ public void start(Stage stage) {
         witness = new MockExceptionWitness(eventBus, witnessed);
         final ExceptionWitnessResponderButton button = new ExceptionWitnessResponderButton(witnessed, "Test Button Popover");
         eventBus.register(button);
-        Scene scene = new Scene(button, 800, 600);
+        Scene scene = new Scene(new Pane(button), 800, 600);
```
Comment (JLLeitschuh) [inline ui/src/test/java/edu/wpi/grip/ui/components/ExceptionWitnessResponderButtonTest.java]: Why did you change this?

File: ui/src/main/java/edu/wpi/grip/ui/pipeline/StepController.java [hunk 2]
```diff
@@ -30,18 +31,13 @@
...
-    private VBox inputs;
-    @FXML
-    private VBox outputs;
+    @FXML private VBox root;
+    @FXML private Labeled title;
+    @FXML private ImageView icon;
+    @FXML private HBox buttons;
+    @FXML private VBox inputs;
+    @FXML private VBox outputs;
+    @FXML private Pane exceptionWitnessPane;
```
Comment (JLLeitschuh) [inline ui/src/main/java/edu/wpi/grip/ui/pipeline/StepController.java]: Why is this here?
This class didn't need to be changed at all.

**Comment by JLLeitschuh** [comment]
The resizing is weird and somewhat sickening. Can you just leave it so that it flashes.
Aso, the button was square before. I don't know why its wider now but it shouldn't be.

File: ui/src/main/java/edu/wpi/grip/ui/components/ExceptionWitnessResponderButton.java [hunk 3]
```diff
@@ -127,22 +124,40 @@ private void assignFromExceptionEvent(ExceptionEvent event) {
...
+        icon.setFitHeight(DPIUtility.SMALL_ICON_SIZE);
+
+        FadeTransition ft = new FadeTransition(Duration.millis(500), icon);
+        ft.setToValue(0.5);
+        ft.setCycleCount(Transition.INDEFINITE);
+        ft.setAutoReverse(true);
+        ft.play();
 
-        this.setVisible(false);
+        ScaleTransition st = new ScaleTransition(Duration.millis(500), icon);
```
Comment (JLLeitschuh) [inline ui/src/main/java/edu/wpi/grip/ui/components/ExceptionWitnessResponderButton.java]: This is somewhat sickening to me. Please remove it.

File: ui/src/main/java/edu/wpi/grip/ui/components/ExceptionWitnessResponderButton.java [hunk 4]
```diff
@@ -63,12 +59,12 @@
         private final TitledPane stackTracePane;
 
         private ExceptionPopOver(String title) {
-            super();
```
Comment (JLLeitschuh) [inline ui/src/main/java/edu/wpi/grip/ui/components/ExceptionWitnessResponderButton.java]: Why was this removed?

Comment (ThomasJClark) [inline ui/src/main/java/edu/wpi/grip/ui/components/ExceptionWitnessResponderButton.java]: Not sure

File: ui/src/test/java/edu/wpi/grip/ui/components/ExceptionWitnessResponderButtonTest.java (hunk 1 above)
Comment (ThomasJClark) [inline ui/src/test/java/edu/wpi/grip/ui/components/ExceptionWitnessResponderButtonTest.java]: k

File: ui/src/main/java/edu/wpi/grip/ui/components/ExceptionWitnessResponderButton.java (hunk 4 above)
Comment (ThomasJClark) [inline ui/src/main/java/edu/wpi/grip/ui/components/ExceptionWitnessResponderButton.java]: Although, to be fair, I could ask why it was there in the first place.

File: ui/src/main/java/edu/wpi/grip/ui/components/ExceptionWitnessResponderButton.java [hunk 5]
```diff
@@ -127,22 +126,33 @@ private void assignFromExceptionEvent(ExceptionEvent event) {
...
+
+        setOnMouseClicked(event -> getPopover().show(this));
+        setVisible(false);
+        setContentDisplay(ContentDisplay.GRAPHIC_ONLY);
+        setGraphic(icon);
+        icon.setFitWidth(DPIUtility.SMALL_ICON_SIZE);
+        icon.setFitHeight(DPIUtility.SMALL_ICON_SIZE);
 
-        this.setVisible(false);
+        FadeTransition ft = new FadeTransition(Duration.millis(500), icon);
```
Comment (JLLeitschuh) [inline ui/src/main/java/edu/wpi/grip/ui/components/ExceptionWitnessResponderButton.java]: Why did you reduce this in time? It was 750 originally. It made it look smoother.
To be honest why did you modify the fade transition code at all? It looked fine the way it was.

Comment (ThomasJClark) [inline ui/src/main/java/edu/wpi/grip/ui/components/ExceptionWitnessResponderButton.java]: It looks weird at 750, but sure.  I'll change it to that.

**Comment by JLLeitschuh** [comment]
The button still isn't square. Its a rectangle.

Alert Button

**Comment by ThomasJClark** [comment]
So? None of the buttons have ever been square. I don't remember that ever being a requirement

**Comment by JLLeitschuh** [comment]
It looks better square. Can you do it or is it not possible?

**Comment by ThomasJClark** [comment]
I honestly don't care enough to even try, but it's probably possible to make it perfectly square.

**Comment by JLLeitschuh** [comment]
Is that a `Wontfix`?

**Comment by ThomasJClark** [comment]
Yeah, sure

**Pull request merged**